src/wp-admin/includes/network.php

```diff
@@ -138,7 +138,10 @@ function network_step1( $errors = false ) {
 
 	$hostname  = get_clean_basedomain();
 	$has_ports = strstr( $hostname, ':' );
-	if ( ( false !== $has_ports && ! in_array( $has_ports, array( ':80', ':443' ), true ) ) ) {
+	/** This filter is documented in wp-includes/ms-settings.php */
+	$allowed_ports = apply_filters( 'allowed_multisite_ports', array( ':80', ':443' ) );
+
+	if ( ( false !== $has_ports && ! in_array( $has_ports, $allowed_ports ) ) ) {
```

Comment:
```suggestion
	if ( ( false !== $has_ports && is_array( $allowed_ports ) && ! in_array( $has_ports, $allowed_ports, true ) ) ) {
```
- We call `in_array()` without confirming that `$allowed_ports` is an array after filtering. We should add this guard.
- Adding `true` to `in_array()` will make the check strict. `in_array()` should perform a strict check where possible.